File: ios/SensieModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SensieModule, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(add:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(subtract:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(whipCounter:(NSDictionary)p
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(evaluateSensie:(NSDictionary)sensie withSensies:(NSDictionaryArray)sensies
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(signalStrength:(NSDictionaryArray)sensies
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/SensieModule.swift
import Foundation
import SensieFramework
@objc(SensieModule)
class SensieModule: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    } // for testing
    @objc(add:withB:withResolver:withRejecter:)
    func add(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a+b)
    } // for testing
    @objc(subtract:withB:withResolver:withRejecter:)
    func subtract(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a-b)
    } // for testing
    
    @objc(whipCounter:withResolver:withRejecter:)
    func whipCounter(p: NSDictionary, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(SensieFramework.whipCounter(param: p))
    }
    @objc(evaluateSensie:withSensies:withResolver:withRejecter:)
    func evaluateSensie(sensie: NSDictionary, sensies: [NSDictionary], resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        guard let sensieDic = sensie as? [String: Any],
              let sensiesDic = sensies as? [[String: Any]] else {
            return
        }
        resolve(SensieFramework.evaluateSensie(sensie: sensieDic, sensies: sensiesDic))
    }
    @objc(signalStrength:withResolver:withRejecter:)
    func signalStrength(sensies: [NSDictionary], resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        guard let sensiesDic = sensies as? [[String: Any]] else {
            return
        }
        resolve(SensieFramework.signalStrength(sensies: sensiesDic))
    }
}
